### ISSUE TITLE
[receiver / postgres] Postgresql metrics receiver has runtime error 

### ISSUE THREAD

**Issue by Alex-Garg** [open-telemetry/opentelemetry-collector-contrib#16769]
### Component(s)

receiver/postgresql

## Description
Postgres metrics receiver has runtime error when executing the request on pg_stat_replication (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9897fc5080eb2a246fd85debf95ff83ba02f3670/receiver/postgresqlreceiver/client.go#L458)

## Steps to Reproduce
Configure otelcol to receiver metrics from a Postgres 14.6 

## Actual Result
`scraperhelper/scrapercontroller.go:214  Error scraping metrics  {"kind": "receiver", "name": "postgresql", "pipeline": "metrics", "error": "unable to query pg_stat_replication: pq: COALESCE types interval and integer cannot be matched", "scraper": "postgresql"}`

Request executed directly on postgres : 
```
postgres=# SELECT
        client_addr,
        coalesce(pg_wal_lsn_diff(pg_current_wal_lsn(), replay_lsn), -1) AS replication_bytes_pending,
        coalesce(write_lag, -1),
        coalesce(flush_lag, -1),
        coalesce(replay_lag, -1)
        FROM pg_stat_replication;
ERROR:  COALESCE types interval and integer cannot be matched
LINE 4:  coalesce(write_lag, -1),
                             ^
```

### Collector version

0.66.0

## Environment
OS: CentOS 7 + Docker containers

### OpenTelemetry Collector configuration

```yaml
receivers:
  opencensus:
  otlp:
    protocols:
      grpc:
  postgresql:
    endpoint: postgres:5432
    transport: tcp
    username: postgres
    password: postgres
    databases:
      - db1
      - db2
    collection_interval: 10s
    tls:
      insecure: true
```

### Log output

```shell
scraperhelper/scrapercontroller.go:214  Error scraping metrics  {"kind": "receiver", "name": "postgresql", "pipeline": "metrics", "error": "unable to query pg_stat_replication: pq: COALESCE types interval and integer cannot be matched", "scraper": "postgresql"}
```

**Comment by djaglowski** [comment]
@dehaansa, do you see a proper solution here?

**Comment by dehaansa** [comment]
I'm kind of at a loss at the moment, I've tested against an older version & checked documentation and there's no way that this query should have worked previously.

We'll need to correct the query to do the following for the 3 interval fields ```extract('epoch' from coalesce(field, '-1 seconds'))``` which should then resolve the issue. I'll throw together a quick PR but I don't have time to fully validate at the moment.

**Comment by djaglowski** [comment]
Thanks for taking a look at it, Sam.

@Alex-Garg, is there any chance you have the capability to build the collector locally from #16845 in order to manually test the change?

**Comment by wfhartford** [comment]
I have also been experiencing this issue. I built a collector with the PR's postgresql receiver and no longer get the error message logged. I don't have a PG cluster doing replication, so cannot confirm that the appropriate metrics are being collected.

**Comment by djaglowski** [comment]
I believe this issue was fixed by #16845.
